src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/Utilities/NameGenerator.cs

```diff
@@ -21,24 +21,30 @@ internal static class NameGenerator
             IEnumerable<string> reservedNames,
             bool isCaseSensitive = true)
         {
-            using var namesDisposer = ArrayBuilder<string>.GetInstance(out var names);
-            using var isFixedDisposer = ArrayBuilder<bool>.GetInstance(out var isFixed);
-            using var nameSetDisposer = PooledHashSet<string>.GetInstance(out var nameSet);
+            using var _ = PooledHashSet<string>.GetInstance(out var pooledNameSet);
+            HashSet<string> nameSet = pooledNameSet;
 
-            names.Add(baseName);
-            isFixed.Add(false);
+            if (!isCaseSensitive)
+            {
+                // The pooled collections don't allow non-default comparers (and they can't be set on
+                //   dictionary/hashset after creation). As this scenario is infrequent, we'll just
+                //   create allocate new objects for these. The pooled object will still be freed
+                //   via the PooledDisposer.
+                nameSet = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
```

Comment:
you can make a dedicated pool for the ignorecase case.  i'd prefer that.

Reply:
also fine to have our own dedicated string pool for the case sensitive case.

Reply:
Let me know if you'd rather me keep the changes restricted to adding pooling just in this class